### PR TITLE
Added AvatarList features for creating userlists in AC scripts

### PR DESCRIPTION
**Pull request by CrackerHax**
avatarAddedEvent(sessionUUID);
avatarRemovedEvent(sessionUUID);
avatarSessionChanged(sessionUUID, oldsessionUUID);

Usage:
AvatarList.avatarRemovedEvent.connect(foofunction);
AvatarList.avatarSessionChangedEvent.connect(barfunction);
AvatarList.avatarAddedEvent.connect(foobarfunction);

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by howard-stearns** [comment]
Hi @CrackerHax. It looks like this doesn't interact at all with https://github.com/highfidelity/hifi/pull/6154 sorted out, which is nice. The functionality is also handy, as otherwise there's no good way to know when a session/avatar has gone away. But I do want to get 6154 squared away first.

I'll 'ok to test' shortly, but first today I do want to make sure that some of our build machinery has stabilized after some issues last night.

Thanks.

File: assignment-client/src/avatars/AvatarMixer.cpp [hunk 1]
```diff
@@ -254,10 +254,10 @@ void AvatarMixer::broadcastAvatarData() {
                     // potentially update the max full rate distance for this frame
                     maxAvatarDistanceThisFrame = std::max(maxAvatarDistanceThisFrame, distanceToAvatar);
 
-                    if (distanceToAvatar != 0.0f
```
Comment (ZappoMan) [inline assignment-client/src/avatars/AvatarMixer.cpp]: formatting is wrong on this file. Can you please repair... it doesn't look like you changed anything other than white space. maybe you can revert your changes to this file

**Comment by howard-stearns** [comment]
ok to test

**Comment by howard-stearns** [comment]
test this please

**Comment by howard-stearns** [comment]
@murillodigital, how do I get gustavo to make a build for testing?

**Comment by ZappoMan** [comment]
ok to test

**Comment by hifi-gustavo** [comment]
Build console output for commit: d34fc35544fa38715d1990f0e09f3201c1f68679 available at: https://gist.github.com/81e4e44f9cb6e595e609<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6195.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6195.dmg

**Comment by howard-stearns** [comment]
Test out ok. We can fix up the whitespace later. (Please always use 4-space/no-tabs, as the uniformity helps us recognize code easier.)

File: assignment-client/src/avatars/AvatarMixer.cpp (hunk 1 above)
Comment (CrackerHax) [inline assignment-client/src/avatars/AvatarMixer.cpp]: I'm sure it will get fixed when somebody posts a paying job for it.

**Pull request merged**